teku/src/test/java/tech/pegasys/teku/cli/BeaconNodeCommandTest.java

```diff
@@ -717,12 +718,21 @@ private void assertTekuAndLoggingConfiguration(
       final TekuConfiguration expected, final LoggingConfig expectedLogging) {
     assertTekuConfiguration(expected);
     final LoggingConfig actualLogging = getResultingLoggingConfiguration();
-    assertThat(actualLogging).usingRecursiveComparison().isEqualTo(expectedLogging);
+    assertThat(actualLogging).isEqualTo(expectedLogging);
   }
 
   private void assertTekuConfiguration(final TekuConfiguration expected) {
     final TekuConfiguration actual = getResultingTekuConfiguration();
-    assertThat(actual).usingRecursiveComparison().isEqualTo(expected);
+
+    // Assert Spec object separately to avoid recursion issues
+    assertThat(actual.eth2NetworkConfiguration().getSpec())
+        .isEqualTo(expected.eth2NetworkConfiguration().getSpec());
+
+    // Ignore any Spec assertion on recursion
+    assertThat(actual)
+        .usingRecursiveComparison()
+        .ignoringFieldsOfTypes(Spec.class)
+        .isEqualTo(expected);
```

Comment:
Completely removing this recursive check would require too much work. I have used a selective assertion strategy to minimize the performance impact while still checking that the config matches the expected values.